docs/spec/staking/state.md

```diff
@@ -152,29 +149,26 @@ type UnbondingDelegationEntry struct {
 
 ### Redelegation
 
-Shares in a `Delegation` can be rebonded to a different validator, but they must
-for some time exist as a `Redelegation`, where shares can be reduced if Byzantine
-behavior is detected. This is tracked as moving a delegation from a `ValidatorSrcAddr`
-to a `ValidatorDstAddr`.
+The bonded tokens worth of a `Delegation` may be instantly redelegated from a
+source validator to a different validator (destination validator). However when
+this occurs they must be tracked in a `Redelegation` object, whereby their
+shares can be slashed if their tokens were found to contribute to Byzantine
+behavior in the source validator.  This is tracked as moving a delegation from
+a `ValidatorSrcAddr` to a `ValidatorDstAddr`.
 
 `Redelegation` are indexed in the store as:
 
- - Redelegations: `0x0D | DelegatorAddr | ValidatorSrcAddr | ValidatorDstAddr ->
-   amino(redelegation)`
- - RedelegationsBySrc: `0x0E | ValidatorSrcAddr | ValidatorDstAddr |
-   DelegatorAddr -> nil`
- - RedelegationsByDst: `0x0F | ValidatorDstAddr | ValidatorSrcAddr | DelegatorAddr
-   -> nil`
+ - Redelegations: `0x34 | DelegatorAddr | ValidatorSrcAddr | ValidatorDstAddr -> amino(redelegation)`
+ - RedelegationsBySrc: `0x35 | ValidatorSrcAddr | ValidatorDstAddr | DelegatorAddr -> nil`
+ - RedelegationsByDst: `0x36 | ValidatorDstAddr | ValidatorSrcAddr | DelegatorAddr -> nil`
 
 The first map here is used for queries, to lookup all redelegations for a given
 delegator. The second map is used for slashing based on the `ValidatorSrcAddr`,
-while the third map is for slashing based on the ToValOwnerAddr.
+while the third map is for slashing based on the `ValidatorDstAddr`.
 
 A redelegation object is created every time a redelegation occurs. The
-redelegation must be completed with a second transaction provided by the
-delegation owner after the unbonding period has passed.  The destination
-delegation of a redelegation may not itself undergo a new redelegation until
-the original redelegation has been completed.
+destination delegation of a redelegation may not itself undergo a new
```

Comment:
Confusingly worded. The destination validator could have other redelegations - rather, it's that the delegator cannot redelegate away from a validator to which they are currently redelegating.

Reply:
Agree was confusing, made considerably more clear IMO: 

```
A redelegation object is created every time a redelegation occurs.  To prevent
"redelegation hopping" redelegations may not occure under the situation that:
 - the (re)delegator already has another unmature redelegation in progress
   with a destination to a validator (let's call it `Validator X`)
 - and, the (re)delegator is attempting to create a _new_ redelegation 
   where the source validator for this new redelegation is `Validator-X`. 
```